### ISSUE TITLE
Server freeze on private modpack (1.16.3)

### ISSUE THREAD

**Issue by IDriftI** [Alex-the-666/Ice_and_Fire#3077]
My server will freeze when reaching certain areas. It will go from zero lag to a complete freeze. I have more than enough power and RAM on my pc to accommodate for my mod-pack. It also only happens in certain areas for unknown reasons. I have tested removing ice and fire specifically and that fixed it, but I'd rather keep this great mod. 

Here's the crash report:
---- Minecraft Crash Report ----
// Daisy, daisy...
 
Time: 10/29/20 3:35 PM
Description: Watching Server
 
java.lang.Error: ServerHangWatchdog detected that a single server tick took 60.00 seconds (should be max 0.05)
	at sun.misc.Unsafe.park(Native Method) ~[?:1.8.0_251] {}
	at java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source) ~[?:1.8.0_251] {}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_223705_bi(SourceFile:139) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:129) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(SourceFile:141) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_222868_e(SourceFile:124) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$$Lambda$7902/1603000793.get(Unknown Source) ~[?:?] {}
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source) ~[?:1.8.0_251] {}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213166_h(SourceFile:545) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213168_p(SourceFile:554) ~[?:?] {re:classloading}
	at net.minecraft.world.server.ServerChunkProvider.func_217234_d(SourceFile:280) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:718) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:707) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:693) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.lambda$func_240784_a_$0(MinecraftServer.java:229) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer$$Lambda$6611/624129170.run(Unknown Source) ~[?:?] {}
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_251] {}
 
 
A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------
 
-- Head --
Thread: Server Watchdog
Stacktrace:
	at sun.misc.Unsafe.park(Native Method) ~[?:1.8.0_251] {}
	at java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source) ~[?:1.8.0_251] {}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_223705_bi(SourceFile:139) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:129) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(SourceFile:141) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider.func_222868_e(SourceFile:124) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$$Lambda$7902/1603000793.get(Unknown Source) ~[?:?] {}
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source) ~[?:1.8.0_251] {}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213166_h(SourceFile:545) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213168_p(SourceFile:554) ~[?:?] {re:classloading}
	at net.minecraft.world.server.ServerChunkProvider.func_217234_d(SourceFile:280) ~[?:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:718) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:707) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:693) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.lambda$func_240784_a_$0(MinecraftServer.java:229) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
-- Thread Dump --
Details:
	Threads: "Worker-Main-33" Id=102 WAITING on java.util.concurrent.CompletableFuture$Signaller@310beeff
	at sun.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.CompletableFuture$Signaller@310beeff
	at java.util.concurrent.locks.LockSupport.park(Unknown Source)
	at java.util.concurrent.CompletableFuture$Signaller.block(Unknown Source)
	at java.util.concurrent.ForkJoinPool.managedBlock(Unknown Source)
	at java.util.concurrent.CompletableFuture.waitingGet(Unknown Source)
	at java.util.concurrent.CompletableFuture.join(Unknown Source)
	at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(SourceFile:124)
	at net.minecraft.world.World.func_217353_a(World.java:167)
	...
 
 
"Netty Server IO #2" Id=95 RUNNABLE (in native)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(Unknown Source)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(Unknown Source)
	at sun.nio.ch.WindowsSelectorImpl.doSelect(Unknown Source)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(Unknown Source)
	-  locked io.netty.channel.nio.SelectedSelectionKeySet@35347c06
	-  locked java.util.Collections$UnmodifiableSet@7036e4c4
	-  locked sun.nio.ch.WindowsSelectorImpl@2f1e2a13
	at sun.nio.ch.SelectorImpl.select(Unknown Source)
	at io.netty.channel.nio.SelectedSelectionKeySetSelector.select(SelectedSelectionKeySetSelector.java:62)
	at io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:756)
	...
 
 
"Netty Server IO #1" Id=82 RUNNABLE
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(Unknown Source)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(Unknown Source)
	at sun.nio.ch.WindowsSelectorImpl.doSelect(Unknown Source)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(Unknown Source)
	-  locked io.netty.channel.nio.SelectedSelectionKeySet@418c5519
	-  locked java.util.Collections$UnmodifiableSet@574ec562
	-  locked sun.nio.ch.WindowsSelectorImpl@10191b8b
	at sun.nio.ch.SelectorImpl.select(Unknown Source)
	at io.netty.channel.nio.SelectedSelectionKeySetSelector.select(SelectedSelectionKeySetSelector.java:62)
	at io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:756)
	...
 
 
"Server Watchdog" Id=78 RUNNABLE
	at sun.management.ThreadImpl.dumpThreads0(Native Method)
	at sun.management.ThreadImpl.dumpAllThreads(Unknown Source)
	at net.minecraft.server.dedicated.ServerHangWatchdog.run(ServerHangWatchdog.java:37)
	at java.lang.Thread.run(Unknown Source)
 
 
"Thread-17" Id=72 RUNNABLE (in native)
	at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus0(Native Method)
	at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus(Unknown Source)
	at sun.nio.fs.WindowsWatchService$Poller.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
 
 
"Netty Server IO #0" Id=71 RUNNABLE (in native)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(Unknown Source)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(Unknown Source)
	at sun.nio.ch.WindowsSelectorImpl.doSelect(Unknown Source)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(Unknown Source)
	-  locked io.netty.channel.nio.SelectedSelectionKeySet@4e09e529
	-  locked java.util.Collections$UnmodifiableSet@6d0e556c
	-  locked sun.nio.ch.WindowsSelectorImpl@4c4ae87a
	at sun.nio.ch.SelectorImpl.select(Unknown Source)
	at io.netty.channel.nio.SelectedSelectionKeySetSelector.select(SelectedSelectionKeySetSelector.java:62)
	at io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:756)
	...
 
 
"ObjectCleanerThread" Id=70 TIMED_WAITING on java.lang.ref.ReferenceQueue$Lock@1457673b
	at java.lang.Object.wait(Native Method)
	-  waiting on java.lang.ref.ReferenceQueue$Lock@1457673b
	at java.lang.ref.ReferenceQueue.remove(Unknown Source)
	at io.netty.util.internal.ObjectCleaner$1.run(ObjectCleaner.java:54)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Unknown Source)
 
 
"Server console handler" Id=69 RUNNABLE (in native)
	at java.io.FileInputStream.readBytes(Native Method)
	at java.io.FileInputStream.read(Unknown Source)
	at java.io.BufferedInputStream.read1(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	-  locked java.io.BufferedInputStream@1576f835
	at sun.nio.cs.StreamDecoder.readBytes(Unknown Source)
	at sun.nio.cs.StreamDecoder.implRead(Unknown Source)
	at sun.nio.cs.StreamDecoder.read(Unknown Source)
	-  locked java.io.InputStreamReader@acc7ac2
	at java.io.InputStreamReader.read(Unknown Source)
	...
 
 
"DestroyJavaVM" Id=68 RUNNABLE
 
 
"Server thread" Id=65 TIMED_WAITING on java.lang.String@11de5ad9
	at sun.misc.Unsafe.park(Native Method)
	-  waiting on java.lang.String@11de5ad9
	at java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source)
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_223705_bi(SourceFile:139)
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:129)
	at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(SourceFile:141)
	at net.minecraft.world.server.ServerChunkProvider.func_222868_e(SourceFile:124)
	at net.minecraft.world.server.ServerChunkProvider$$Lambda$7902/1603000793.get(Unknown Source)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	...
 
 
"Snooper Timer" Id=66 WAITING on java.util.TaskQueue@3e9f9966
	at java.lang.Object.wait(Native Method)
	-  waiting on java.util.TaskQueue@3e9f9966
	at java.lang.Object.wait(Unknown Source)
	at java.util.TimerThread.mainLoop(Unknown Source)
	at java.util.TimerThread.run(Unknown Source)
 
 
"Thread-3" Id=35 RUNNABLE (in native)
	at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus0(Native Method)
	at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus(Unknown Source)
	at sun.nio.fs.WindowsWatchService$Poller.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
 
 
"Timer hack thread" Id=26 TIMED_WAITING
	at java.lang.Thread.sleep(Native Method)
	at net.minecraft.util.Util$6.run(SourceFile:636)
 
 
"Thread-2" Id=17 RUNNABLE (in native)
	at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus0(Native Method)
	at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus(Unknown Source)
	at sun.nio.fs.WindowsWatchService$Poller.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
 
 
"Thread-1" Id=16 TIMED_WAITING
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source)
	at com.electronwill.nightconfig.core.file.FileWatcher$WatcherThread.run(FileWatcher.java:190)
 
 
"Attach Listener" Id=5 RUNNABLE
 
 
"Signal Dispatcher" Id=4 RUNNABLE
 
 
"Finalizer" Id=3 WAITING on java.lang.ref.ReferenceQueue$Lock@1c0a69d0
	at java.lang.Object.wait(Native Method)
	-  waiting on java.lang.ref.ReferenceQueue$Lock@1c0a69d0
	at java.lang.ref.ReferenceQueue.remove(Unknown Source)
	at java.lang.ref.ReferenceQueue.remove(Unknown Source)
	at java.lang.ref.Finalizer$FinalizerThread.run(Unknown Source)
 
 
"Reference Handler" Id=2 WAITING on java.lang.ref.Reference$Lock@75f90af9
	at java.lang.Object.wait(Native Method)
	-  waiting on java.lang.ref.Reference$Lock@75f90af9
	at java.lang.Object.wait(Unknown Source)
	at java.lang.ref.Reference.tryHandlePending(Unknown Source)
	at java.lang.ref.Reference$ReferenceHandler.run(Unknown Source)
 
 
 
Stacktrace:
	at net.minecraft.server.dedicated.ServerHangWatchdog.run(ServerHangWatchdog.java:52) [?:?] {re:classloading}
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_251] {}
 
 
-- System Details --
Details:
	Minecraft Version: 1.16.3
	Minecraft Version ID: 1.16.3
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_251, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 4576563856 bytes (4364 MB) / 5924978688 bytes (5650 MB) up to 5924978688 bytes (5650 MB)
	CPUs: 8
	JVM Flags: 2 total; -Xmx6G -Xms4G
	ModLauncher: 7.0.1+78+master.e9771d8
	ModLauncher launch target: fmlserver
	ModLauncher naming: srg
	ModLauncher services: 
		/mixin-0.8.2.jar mixin PLUGINSERVICE 
		/eventbus-3.0.3-service.jar eventbus PLUGINSERVICE 
		/forge-1.16.3-34.1.25.jar object_holder_definalize PLUGINSERVICE 
		/forge-1.16.3-34.1.25.jar runtime_enum_extender PLUGINSERVICE 
		/accesstransformers-2.2.0-shadowed.jar accesstransformer PLUGINSERVICE 
		/forge-1.16.3-34.1.25.jar capability_inject_definalize PLUGINSERVICE 
		/forge-1.16.3-34.1.25.jar runtimedistcleaner PLUGINSERVICE 
		/mixin-0.8.2.jar mixin TRANSFORMATIONSERVICE 
		/forge-1.16.3-34.1.25.jar fml TRANSFORMATIONSERVICE 
	FML: 34.1
	Forge: net.minecraftforge:34.1.25
	FML Language Providers: 
		javafml@34.1
		minecraft@1
	Mod List: 
		forge-1.16.3-34.1.25-server.jar Minecraft {minecraft@1.16.3 DONE}
		useful_backpacks-1.16.3-1.11.6.83.jar Useful Backpacks {usefulbackpacks@1.11.6.83 DONE}
		TrashSlot_1.16.3-12.2.0.jar TrashSlot {trashslot@12.2.0 DONE}
		NetherPortalFix_1.16.3-7.2.0.jar NetherPortalFix {netherportalfix@7.2.0 DONE}
		ImmersiveEngineering-1.16.3-4.1.0-124.jar Immersive Engineering {immersiveengineering@1.16.3-4.1.0-124 DONE}
		BiomesOPlenty-1.16.3-12.0.0.409-universal.jar Biomes O' Plenty {biomesoplenty@1.16.3-12.0.0.409 DONE}
		valhelsia_structures-16.0.5.jar Valhelsia Structures {valhelsia_structures@16.0.5 DONE}
		pamhc2crops-1.16.3-1.0.1.jar Pam's HarvestCraft 2 Crops {pamhc2crops@version DONE}
		jei-1.16.3-7.6.0.49.jar Just Enough Items {jei@7.5.0.43 DONE}
		AttributeFix-1.16.3-8.0.1.jar AttributeFix {attributefix@8.0.1 DONE}
		Patchouli-1.16.2-47.jar Patchouli {patchouli@1.16.2-47 DONE}
		BetterCaves-1.16.2-1.0.3.jar YUNG's Better Caves {bettercaves@1.16.2-1.0.3 DONE}
		Xaeros_Minimap_20.26.0_Forge_1.16.3.jar Xaero's Minimap {xaerominimap@20.26.0 DONE}
		Waystones_1.16.3-7.3.0.jar Waystones {waystones@7.3.0 DONE}
		pamhc2foodcore-1.16.3-1.0.0.jar Pam's HarvestCraft 2 Food Core {pamhc2foodcore@version DONE}
		gravestone-1.16.3-2.0.3.jar Gravestone Mod {gravestone@1.16.3-2.0.3 DONE}
		Clumps-6.0.0.11.jar Clumps {clumps@6.0.0.11 DONE}
		XaerosWorldMap_1.11.0_Forge_1.16.3.jar Xaero's World Map {xaeroworldmap@1.11.0 DONE}
		CookingForBlockheads_1.16.3-9.2.1.jar Cooking for Blockheads {cookingforblockheads@9.2.1 DONE}
		Placebo-1.16.3-4.3.2.jar Placebo {placebo@4.3.2 DONE}
		comforts-forge-1.16.3-4.0.0.1.jar Comforts {comforts@1.16.3-4.0.0.1 DONE}
		citadel-1.4.1.jar Citadel {citadel@1.4.1 DONE}
		rats-7.0.0-1.16.3.jar Rats {rats@7.0.0 DONE}
		AutoRegLib-1.6-46 (1).jar AutoRegLib {autoreglib@1.6-46 DONE}
		culinaryconstruct-forge-1.16.3-4.0.0.1.jar Culinary Construct {culinaryconstruct@1.16.3-4.0.0.1 DONE}
		reliquary-1.16.3-1.3.4.1042.jar Reliquary {xreliquary@1.16.3-1.3.4.1042 DONE}
		u_team_core-1.16.3-3.1.10.183.jar U Team Core {uteamcore@3.1.10.183 DONE}
		iceandfire-2.1.0-1.16.3.jar Ice and Fire {iceandfire@2.0.3-1.16.3 DONE}
		solcarrot-1.16.3-1.9.8.jar Spice of Life: Carrot Edition {solcarrot@1.16.3-1.9.8 DONE}
		Hwyla-forge-1.10.11-B78_1.16.2.jar Waila {waila@1.10.11-B78_1.16.2 DONE}
		forge-1.16.3-34.1.25-universal.jar Forge {forge@34.1.25 DONE}
		Psi1.16-87.jar Psi {psi@1.16-87 DONE}
		Quark-r2.4-269.jar Quark {quark@r2.4-269 DONE}
		Apotheosis-1.16.3-4.3.4.jar Apotheosis {apotheosis@4.3.4 DONE}
		AppleSkin-mc1.16.2-forge-1.0.14.jar AppleSkin {appleskin@1.0.14 DONE}
		Aquaculture-1.16.3-2.1.1.jar Aquaculture 2 {aquaculture@1.16.3-2.1.1 DONE}
		ironchest-1.16.3-11.2.6.jar Iron Chests {ironchest@1.16.3-11.2.6 DONE}
		ratlantis-1.0.0-1.16.3.jar Rats: Ratlantis {ratlantis@1.0.0-1.16.3 DONE}
	Crash Report UUID: 6a5a2b7d-7c3c-4a52-8f80-3aef572187bd
	[Psi] Active spell: None
	Player Count: 1 / 20; [ServerPlayerEntity['_Drift_'/485, l='ServerLevel[world]', x=2448.16, y=85.77, z=-6144.09]]
	Data Packs: vanilla, mod:usefulbackpacks, mod:trashslot (incompatible), mod:netherportalfix (incompatible), mod:immersiveengineering (incompatible), mod:biomesoplenty (incompatible), mod:pamhc2crops (incompatible), mod:jei, mod:attributefix, mod:bettercaves (incompatible), mod:xaerominimap, mod:waystones (incompatible), mod:pamhc2foodcore (incompatible), mod:gravestone, mod:clumps (incompatible), mod:xaeroworldmap, mod:cookingforblockheads (incompatible), mod:placebo (incompatible), mod:comforts (incompatible), mod:citadel (incompatible), mod:rats, mod:autoreglib (incompatible), mod:culinaryconstruct (incompatible), mod:xreliquary, mod:uteamcore, mod:iceandfire (incompatible), mod:solcarrot (incompatible), mod:waila, mod:forge, mod:quark (incompatible), mod:apotheosis (incompatible), mod:appleskin, mod:aquaculture (incompatible), mod:ironchest (incompatible), mod:ratlantis, mod:valhelsia_structures (incompatible), mod:patchouli (incompatible), mod:psi (incompatible)
	Is Modded: Definitely; Server brand changed to 'forge'
	Type: Dedicated Server (map_server.txt)

**Comment by Twiggeh** [comment]
I have the same issue, the problem seems to be under ground dragons. Keeping those loaded /loading them in makes the server display the "Can't keep up" error message and then it hangs up. Error occurs with java 8 & 11.

[crash-2020-10-31_01.39.40-server.txt](https://github.com/Alex-the-666/Ice_and_Fire/files/5468929/crash-2020-10-31_01.39.40-server.txt)

**Comment by theoretischer** [comment]
same -.-

**Comment by TheBv** [comment]
This issue should've been resolved in the newest Ice and Fire version (2.1.5) if it hasn't let me know.
If there's a similar issue pre 1.16 you could try increase max-tick-time in the server properties. This won't fix that the server will be laggy but it won't crash as soon.

**Comment by Hobnob93** [comment]
> This issue should've been resolved in the newest Ice and Fire version (2.1.5) if it hasn't let me know.
> If there's a similar issue pre 1.16 you could try increase max-tick-time in the server properties. This won't fix that the server will be laggy but it won't crash as soon.

TheBv, hello there! So, with my experience, it seems the issue is fixed in new worlds. If I generate a new world with the same seed as my existing world, then travel to the area that usually hangs, the Mausoleum spawns fine. However, on my existing world with the latest version (2.1.5), the server hang still occurs. Disabling Mausoleums in the config appears to have no effect for this existing world.

Are you able to advise, by any chance? I can send you the crash report, but it appears to be pretty much the same as the one listed above. I have also tried increasing the max-tick-time, but it just crashes after that time, instead of the original 60 seconds.

Thank you for your time, and I appreciate the work you do! As a game and web dev, I feel your pain :D

**Comment by TheBv** [comment]
I assumed that the issue mentioned here was related to dragons causing freezing issues but I guess it could also be the mausoleum. In case if it's the mausoleum the max-tick-time doesn't help at all ^^"
I think there reason why the server still has issues is because it pre-generated the chunk already. There's two steps before chunks get loaded the pregen mod calls them "Main" and "Light" 

>  "Main" is the main generation step, "Light" is the finalization of the chunks so they can be used 

And in the "Main" step the issue with the generation occurs. So I'm assuming that your word already has generated the structure in the "Main"-chunk but since you're not close enough the server crash doesn't happen until the "Light" generation.
Honestly I have no idea how to properly fix that. You could try deleting the chunks somehow?
There's also what Won-Ton mentioned here #3390
But when I don't expand the BoundingBox the structure doesn't generate properly (e.g only a quarter gets generated and similar issues) 
So I think what he mentioned shouldn't occur anymore since after the hotfix #3359 but maybe I'm wrong.

Hope this at least gives some insights!

**Comment by Hobnob93** [comment]
> I assumed that the issue mentioned here was related to dragons causing freezing issues but I guess it could also be the mausoleum. In case if it's the mausoleum the max-tick-time doesn't help at all ^^"
> I think there reason why the server still has issues is because it pre-generated the chunk already. There's two steps before chunks get loaded the pregen mod calls them "Main" and "Light"
> 
> > "Main" is the main generation step, "Light" is the finalization of the chunks so they can be used
> 
> And in the "Main" step the issue with the generation occurs. So I'm assuming that your word already has generated the structure in the "Main"-chunk but since you're not close enough the server crash doesn't happen until the "Light" generation.
> Honestly I have no idea how to properly fix that. You could try deleting the chunks somehow?
> There's also what Won-Ton mentioned here #3390
> But when I don't expand the BoundingBox the structure doesn't generate properly (e.g only a quarter gets generated and similar issues)
> So I think what he mentioned shouldn't occur anymore since after the hotfix #3359 but maybe I'm wrong.
> 
> Hope this at least gives some insights!

Apologies, I thought I was in a completely different issue! Had a few tabs open and must've got confused.

Thank you for your detailed answer, it is much appreciated! I have tried using WorldEdit to remove the problem chunks, but I think it tries to complete the generation before deleting it, as it still hangs. Will see if there are other ways I can go about deleting the chunks.

Thank you!
